ark/utils/deepcell_service_utils.py

```diff
@@ -37,25 +39,39 @@ def create_deepcell_output(deepcell_input_dir, deepcell_output_dir, fovs=None,
             ValueError:
                 Raised if there is some fov X (from fovs list) s.t.
                 the file <deepcell_input_dir>/fovX.tif does not exist
-        """
+    """
+
+    # extract all the files from deepcell_input_dir
+    input_files = io_utils.list_files(deepcell_input_dir, substrs='.tif')
+
+    # set fovs equal to input_files it not already set
     if fovs is None:
-        tifs = io_utils.list_files(deepcell_input_dir, substrs='.tif')
-        fovs = io_utils.extract_delimited_names(tifs, delimiter='.')
+        fovs = input_files
+
+    # now extract only the names of the fovs without the file extension
+    fovs = io_utils.extract_delimited_names(fovs, delimiter='.')
 
+    # make sure that all fovs actually exist in the list of input_files
+    misc_utils.verify_in_list(
+        fovs=fovs,
+        deepcell_input_files=io_utils.extract_delimited_names(input_files, delimiter='.'))
+
+    # define the location of the zip file for our fovs
     zip_path = os.path.join(deepcell_input_dir, 'fovs.zip')
     if os.path.isfile(zip_path):
         warnings.warn(f'{zip_path} will be overwritten.')
 
+    # write all files to the zip file
     print('Zipping preprocessed tif files.')
+
     with ZipFile(zip_path, 'w') as zipObj:
         for fov in fovs:
             filename = os.path.join(deepcell_input_dir, fov + '.tif')
-            if not os.path.isfile(filename):
-                raise ValueError('Could not find .tif file for %s. '
-                                 'Invalid value for %s' % (fov, filename))
             zipObj.write(filename, os.path.basename(filename))
 
-    print('Uploading files to DeppCell server.')
+    # pass the zip file to deepcell.org, then remove it afterwards
+    print('Uploading files to DeepCell server.')
+
```

Comment:
Rather than removing this zip file (line 76), I think we should check if it already exists when creating the zip file above, and then give a warning that it's being overwritten. given the current issues with zip file sizes, and purely for debugging purposes, I think it's useful to keep this zip file around.

Reply:
We already have this warning, so I'll just remove the line that removes the zip file.